code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -228,7 +165,62 @@
 			N.adjustBruteLoss(-1)
 			N.adjustFireLoss(-1)
 
+/obj/item/reagent_containers/food/snacks/sliceable/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Alt-click to put something small inside.</span>")
+
+/obj/item/reagent_containers/food/snacks/sliceable/AltClick(mob/user)
+	var/obj/item/I = user.get_active_hand()
+	if(!I)
+		return
+	if(I.w_class > WEIGHT_CLASS_SMALL)
+		to_chat(user, "<span class='warning'>You cannot fit [I] in [src]!</span>")
+		return
+	var/newweight = GetTotalContentsWeight() + I.GetTotalContentsWeight() + I.w_class
+	if(newweight > MAX_WEIGHT_CLASS)
+		// Nope, no bluespace slice food
+		to_chat(user, "<span class='warning'>You cannot fit [I] in [src]!</span>")
+		return 1
+	if(!iscarbon(user))
+		return 1
+	to_chat(user, "<span class='warning'>You slip [I] inside [src].</span>")
+	user.unEquip(I)
+	if((user.client && user.s_active != src))
+		user.client.screen -= I
+	I.dropped(user)
+	total_w_class += I.w_class
+	add_fingerprint(user)
+	contents += I
+	return
+
+/obj/item/reagent_containers/food/snacks/sliceable/attackby(obj/item/I, mob/user, params)
+	..()
```

Comment:
This should be at the end, after all the handling checks - otherwise, you risk a "double tap".